### PR TITLE
Basic openehr terminology support

### PR DESCRIPTION
**Pull request by pieterbos**
Add basic OpenEHR terminology XML files, plus a parser, plus a very simple access model that is **much** simpler than the one specified in the term services specification.

Integrates with the ADL 1.4 -> ADL 2 converter so it generates correct terms. 

Of course with tests both for the terminology support as well as for the ADL 1.4 -> 2 conversion.

This probably will need some extra work later on when we develop the invariants, but it gives a solid basis already and it's hard to determine from the start what we need and don't need there.

### CODE REVIEW

**Review by MattijsK — APPROVED** [review]
Looks good! Indeed not too complex

File: openehr-terminology/build.gradle [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+//        openehr_terminology {
+//            schemaFile = 'openehr_terminology.xsd'
+//            javaPackageName = 'com.nedap.archie.terminology.openehr'
+//        }
+////        openehr_external_terminology {
+////            schemaFile = 'openehr_external_terminologies.xsd'
+////            javaPackageName = 'com.nedap.archie.terminology.external'
+////        }
+//    }
+//}
```
Comment (J3173) [inline openehr-terminology/build.gradle:31]: Are these comments still needed?

**Review by J3173 — CHANGES_REQUESTED** [review]
Nice! Some minor comments.

File: openehr-terminology/src/main/java/com/nedap/archie/terminology/OpenEHRTerminologyAccess.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+                }
+            }
+
+        } catch (JAXBException | IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private synchronized  TerminologyImpl getOrCreateTerminologyById(String issuer, String openEhrId, String externalId) {
```
[the remark this answers is not in the document]
Comment (pieterbos) [reply openehr-terminology/src/main/java/com/nedap/archie/terminology/OpenEHRTerminologyAccess.java:65]: thanks. I had already merged before your review, but I'll create a new PR to fix your comments :) And I'll fix this.

File: openehr-terminology/src/test/java/com/nedap/archie/terminology/OpenEHRTerminologyAccessTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    private OpenEHRTerminologyAccess termAccess;
+
+    @Before
+    public void getInstance() {
+        termAccess = OpenEHRTerminologyAccess.getInstance();//should not throw an exception from parsing
+    }
+
+    @Test
+    public void getCodesFromGroup() {
+        TermCode mass = termAccess.getTerm("openehr", "124", "en");
```
[the remark this answers is not in the document]
Comment (pieterbos) [reply openehr-terminology/src/test/java/com/nedap/archie/terminology/OpenEHRTerminologyAccessTest.java:21]: good catch, I'll write the test.

File: openehr-terminology/build.gradle (hunk 1 above)
Comment (pieterbos) [reply openehr-terminology/build.gradle:31]: If ever the XSD changes, you need this to regenerate, so that's why I left it here.

**Review by J3173 — APPROVED** [review]

**Pull request merged**